src/extensions/diagnose.ts

```diff
@@ -0,0 +1,74 @@
+import { logPrefix } from "./extensionsHelper";
+import { getProjectNumber } from "../getProjectNumber";
+import * as utils from "../utils";
+import * as resourceManager from "../gcp/resourceManager";
+import { listInstances } from "./extensionsApi";
+import { promptOnce } from "../prompt";
+import { logger } from "../logger";
+import { FirebaseError } from "../error";
+
+const SERVICE_AGENT_ROLE = "roles/firebasemods.serviceAgent";
+
+/**
+ * Diagnoses and optionally fixes known issues with project configuration, ex. missing Extensions Service Agent permissions.
+ * @param projectId ID of the project we're querying
+ */
+export async function diagnose(projectId: string): Promise<boolean> {
+  const projectNumber = await getProjectNumber({ projectId });
+  const firexSaProjectId = utils.envOverride(
+    "FIREBASE_EXTENSIONS_SA_PROJECT_ID",
```

Comment:
Lets add the appropriate staging/autopush values for this to go/firex-eng-playbook